src/agent/communicator/include/communicator.hpp

```diff
@@ -2,9 +2,13 @@
 
 #include <ihttp_client.hpp>
 
+#include <config.h>
+#include <logger.hpp>
+
 #include <boost/asio/awaitable.hpp>
 #include <boost/asio/steady_timer.hpp>
 #include <boost/beast/http/status.hpp>
+#include <boost/url.hpp>
```

Comment:
Looks like boost::url and logger now are dependencies with public linkage, the cmake file needs to be modified.